### PR TITLE
Fix tag association in tables

### PR DESCRIPTION
**Pull request by Fabien-Chouteau**
Global tags such as "BOOT_ROM" were only working for the first iteration of a table, and would otherwise be empty.

Using a String to Unbounded_String association rather the String to Tag fixes the issue. As shown in the output of the print_tags test, the impacted tags are no longer vectors of element.

Part of W109-026.

### CODE REVIEW

File: testsuite/tests/basic_symbol_matching/cortex-m/test.py [hunk 1]
```diff
@@ -2,9 +2,12 @@
                                     check_symbols, MemoryMap
 
 memmap = MemoryMap()
-memmap.add('flash', 'ROM', 0, 32 * 1024)
-memmap.add('ram', 'RAM', 0x10000, 32 * 1024)
-memmap.add('ccm', 'RAM', 0x20000, 32 * 1024)
```
Comment (ashleygay) [inline testsuite/tests/basic_symbol_matching/cortex-m/test.py:7]: There is no equivalent `ccm` memory region below, is that expected ?

**Review by ashleygay — COMMENTED** [review]
Two questions (one in a separate comment, sorry !), other than that looks good to me.

File: testsuite/tests/basic_symbol_matching/cortex-m/test.py (hunk 1 above)
Comment (Fabien-Chouteau) [reply testsuite/tests/basic_symbol_matching/cortex-m/test.py:7]: I changed this test from using memory region from a real board to just imaginary regions that better fit the need of the test.

**Review by ashleygay — APPROVED** [review]
Looks good to me, thanks !

**Pull request merged**